### ISSUE TITLE
OpenGL texture issue on Intel Iris Xe Graphics / Windows

### ISSUE THREAD

**Issue by ds-sloth** [TheXTech/TheXTech#859]
### Describe the issue

There is some sort of texture corruption. This affects Discord user Cinnamations on Intel Iris Xe Graphics. The Windows x86_64 build running on Wine (on Zen 3 integrated graphics) is unaffected.

(It seems like the top row is missing, then some columns at some stride?) This could possibly be a non-power-of-two textures defect, though I believe the affected textures seem to be 64x32 / 32x64.

### A type of bug

Native bug

### Does this issue happens always or randomly?

Happens randomly, often

### Version

v1.3.7-beta

### Platform

Windows

### Processor architecture

x86_64 / x64 (64-bit x86)

### Log file (if presented)
```
00:00.001 [Info]: <Application started>
00:00.343 [Info]: TheXTech 1.3.7-beta (branch main, commit d142b5d)
00:00.503 [Info]: Render GL: compiled for SDL 2.30.7, running with SDL 2.30.7
00:00.542 [Info]: Render GL: attempting to create OpenGL Compatibility 3.3+ context...
00:00.602 [Info]: Render GL: successfully initialized OpenGL 3.3 (Profile Compatibility)
00:00.642 [Info]: OpenGL version: 3.3.0 - Build 31.0.101.5333
00:00.680 [Info]: OpenGL renderer: Intel(R) Iris(R) Xe Graphics
00:00.720 [Info]: GLSL version: 3.30 - Build 31.0.101.5333
00:00.747 [Info]: OpenGL video mode: R10 G10 B10 with 24-bit depth buffer
00:00.775 [Info]: RenderGL: max texture size 16384
00:00.803 [Info]: GLProgramObject: will use the native OpenGL compiler for OpenGL ES 2 shaders.
00:00.832 [Info]: GLProgramObject: will use the native OpenGL compiler for OpenGL ES 3 shaders.
00:00.874 [Warning]: GLProgramObject: attempted to register uniform u_step_size of invalid type 8b5e
00:00.920 [Info]: Initialized OpenGL 3.3-P with draw features: logicOp, depth buffer, client-side arrays, NPOT textures, BGRA format, ES2 shaders, ES3 shaders, and framebuffers: game, fb read, multipass, lighting, depth read
00:07.378 [Info]: Will load SFX D:/Stuff i use/Games/FolderGames/Mario/Mario source files for games/smbx/thextech-super-mario-bros-x/sound/dungeon-win.ogg as a multi-music
00:07.630 [Info]: Will load SFX D:/Stuff i use/Games/FolderGames/Mario/Mario source files for games/smbx/thextech-super-mario-bros-x/sound/game-beat.ogg as a multi-music
00:07.791 [Info]: Will load SFX D:/Stuff i use/Games/FolderGames/Mario/Mario source files for games/smbx/thextech-super-mario-bros-x/sound/smw-exit.ogg as a multi-music
00:07.908 [Info]: Will load SFX D:/Stuff i use/Games/FolderGames/Mario/Mario source files for games/smbx/thextech-super-mario-bros-x/sound/sm-explosion.ogg as a multi-music
00:08.144 [Info]: Loaded sound effects: dumped=98; as music=4
01:46.728 [Info]: <Application closed>
```

### Example Case

![image](https://github.com/user-attachments/assets/37ae3b71-722b-49af-99b1-85083235f317)

### Additional context

Reported by Cinnamations on Discord, waiting for additional reports.

**Comment by Wohlstand** [comment]
I do remember it's a similar corruption occurs on the S3 Savage 4 when I ran on my P3-500 machine. The driver is just sucks, and even more it crashes when I attempt to switch between fullscreen and windowed mode on the fly, and on attempt to switch between different video modes. But, with Nvidia MX* it works just awesome, the same Windows XP is. However difference, I ran faulty thing on XP SP0, and working thing on XP SP3.

**Comment by Wohlstand** [comment]
Related to this, I found an article that tells that OEM drivers for Intel Iris Xe are faulty/problematic, they might seriously affect performance if not speak about glitches, etc. https://dzen.ru/a/Zr86EKaUOHC6wYIR

I guess, it's possible to try install the Generic driver from here Source: https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html and try everything again.

**Comment by Wohlstand** [comment]
Another post related to Intel Iris Xe that it's generally faulty: https://www.ixbt.com/news/2020/10/04/iris-xe-intel.html

**Comment by Wohlstand** [comment]
Conclusion: if the Generic driver will fix the problem, then issue will be closed, and any Intel Iris Xe users should be recommended to install Generic driver from Intel's official site. Otherwise we can implement the workaround to run SDL Render by default once Intel Iris Xe got been detected since it's a faulty thing (even major **commercial** projects were affected by this problem!).

**Comment by Wohlstand** [comment]
@sahlaysta, ping to subscribe you. Once I make a workaround, I will let you know so devel versions will have it bundled.

**Comment by Wohlstand** [comment]
@sahlaysta, I made the workaround, so, once CI will finish the build, please take this binary and try on your end: https://builds.wohlsoft.ru/win32/thextech-bin-win64-stable-1.3.7.x.7z
(it's an executable only, just put it over yours, and run. Old one you can backup somewhere).

**Comment by Wohlstand** [comment]
@sahlaysta, recently I updated the tweak, now you can try with the OpenGL enabled and see, will it work correctly or not. At least my buddy tested this well.

**Comment by Wohlstand** [comment]
https://github.com/TheXTech/TheXTech/commit/90e3f412fbd81d6a7498e5650db98c5251f34f53 The new tweak was done and it seems resolves the problem of OpenGL on Intel Iris Xe by full disabling of the depth buffer feature.

**Comment by Wohlstand** [comment]
I will clsoe this once at least one of reporters will verify the latest dev build so OpenGL render no longer fails.

**Comment by sahlaysta** [comment]
No visual issues on OpenGL in this dev version.

<img width="1587" height="931" alt="Image" src="https://github.com/user-attachments/assets/a1419e91-f63c-43d0-bb48-a00850de987f" />

**Comment by Wohlstand** [comment]
> No visual issues on OpenGL in this dev version.
> <img alt="Image" width="1587" height="931" src="https://private-user-images.githubusercontent.com/51031130/467989243-a1419e91-f63c-43d0-bb48-a00850de987f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI4MzYxNjksIm5iZiI6MTc1MjgzNTg2OSwicGF0aCI6Ii81MTAzMTEzMC80Njc5ODkyNDMtYTE0MTllOTEtZjYzYy00M2QwLWJiNDgtYTAwODUwZGU5ODdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE4VDEwNTEwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4MzcxNjhkNGRkNzdkMmY3ODI5OTAyODYyOGZhMjFiY2U5N2UyNjQ1NTEzNmZhYjVjNDY3NzRhMTRmODc1OWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.37Ik5vi81BaPdpGXosIV-ODtsdRNiEAoZXNGl3b3Dwk">

Super! Thank you very much! Enjoy the thing!

P.S. I saw you attempted to mention audio echos. It's a feature of TheXTech, you might want to learn how to use it to make atmospheric sections: https://github.com/TheXTech/TheXTech/wiki/Using-the-Sound-FX-in-your-levels

**Comment by sahlaysta** [comment]
Yeah, I noticed it was only in the house and figured it was intentional.

**Comment by Wohlstand** [comment]
Today, ChaikaWiFika tested on Intel Iris Xe with Linux (Ubuntu 25) and it just works, totally no glitches. This means, the problem is 100% Windows-only.

**Comment by Wohlstand** [comment]
I going to release a hotfix of 1.3.7.1 that includes Intel Iris Xe fixes.

**Comment by ChristianSilvermoon** [comment]
@Wohlstand the fix for this is definitely present in Stable 1.3.7.2 ?

I had a friend encounter what seems to be exactly this issue with this configuration
```
OS           : Windows 11 Home
CPU          : 13th Gen Intel Core i5-13420H
GPU          : Nvidia RTX 3050 Laptop GPU
Game Version : Stable — 1.3.7.2 (#330db2f8)
```

Which does have Iris Xe Graphics.
When setting the system to use the Nvidia GPU the issue does not occur.
